Sources/AsyncAlgorithms/Locking.swift

```diff
@@ -17,100 +17,69 @@
 @_implementationOnly import WinSDK
 #endif
 
-internal struct Lock {
+private final class Lock {
```

Comment:
This is quite wasteful; it poses two allocations instead of just one location managed by an external item